platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -415,6 +415,10 @@ public boolean onGenericMotionEvent(MotionEvent event) {
 
   @Override
   public boolean onHoverEvent(MotionEvent event) {
+    if (mapZoomButtonController == null) {
```

Comment:
What about extracting this block of code into a `private` method? IMO, we should try to keep `public` methods as clean as possible so it reads like pseudocode (a.k.a. no conditionals, `for`s, etc.).